apps/api/src/expenses/expenses.controller.ts

```diff
@@ -14,9 +25,17 @@ export class ExpensesController {
     return await this.expensesService.listExpenses()
   }
 
-  @Public()
   @Post('create-expense')
-  create(@Body() createExpenseDto: CreateExpenseDto) {
+  create(
+    @AuthenticatedUser()
+    user: KeycloakTokenParsed,
+    @Body()
+    createExpenseDto: CreateExpenseDto,
+  ) {
+    if (!user) {
+      throw new UnauthorizedException()
+    }
```

Comment:
That's a bit too verbose if we don't actually use the user.

You might as well just skip the `@Public` decorator to require authenticated user. (the default behavior)

An important check here we should add is to make sure only admins can actually update and delete expenses.

Right now any authenticated user can update or delete any expense. That's actually quite important to be fixed @dpmarkov

Reply:
I see, just to be sure does that snippet represent the admin roles?
```javascript
@Roles({
  roles: [RealmViewSupporters.role, ViewSupporters.role],
  mode: RoleMatchingMode.ANY,
})
```

Reply:
Yes, it's exactly what we need here

Reply:
Sorry for the delay. It's done!